backend/tests/test_alpha_vantage_client.py

```
@@ -214,3 +215,53 @@ def test_no_yfinance_in_health_router(self):
+        assert result is None
+        mock_get.assert_not_called()
+
+    def test_falls_back_to_db_cache(self):
+        from app.services import alpha_vantage_client as av
+
+        fresh = datetime.now(timezone.utc)
+        with patch(
+            "app.services.alpha_vantage_client._read_db_cache",
+            return_value=("2026-07-01", fresh),
+        ):
+            with patch("app.services.alpha_vantage_client.requests.get") as mock_get:
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)